### PR TITLE
API SS4 compatibility

### PR DESCRIPTION
**Pull request by robbieaverill**
cc @phptek

This pull request:

* Adds and implements namespaces
* Updates configuration values where possible to use YAML instead of _config.php
* Updates code style guidelines to PSR-2

Still to come:

* Rename files for PSR-4 (not in this pull request, to help preserve file history)
* Marked \@todo comments in the code

### CODE REVIEW

File: _config/documentconverter.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+Name: cwpdocumentconverterconfig
+Only:
+  moduleexists: silverstripe/documentconverter
+  EnvVarSet: DOCVERT_USERNAME
+---
+Silverstripe\DocumentConverter\DocVertConnector:
```
Comment (NightJar) [inline _config/documentconverter.yml]: During our peer session we changed the classes to drop the Document/docvert/conversion prefixes
So this has been updated to `SilverStripe\DocumentConverter\ServiceConnector`
The rest is all good though :)

File: code/CwpInitialisationFilter.php [hunk 2]
```diff
@@ -1,55 +1,84 @@
...
+    public function preRequest(HTTPRequest $request)
+    {
+        if (Config::inst()->get(__CLASS__, 'egress_proxy_default_enabled')) {
+            if (Environment::getEnv('SS_OUTBOUND_PROXY') && Environment::getEnv('SS_OUTBOUND_PROXY_PORT')) {
+                $proxy = Environment::getEnv('SS_OUTBOUND_PROXY');
+                $proxyPort = Environment::getEnv('SS_OUTBOUND_PROXY_PORT');
 
-		}
+                Environment::setEnv('http_proxy', $proxy . ':' . $proxyPort);
+                Environment::setEnv('https_proxy', $proxy. ':' . $proxyPort);
```
Comment (NightJar) [inline code/CwpInitialisationFilter.php]: Does phpcs not take issue with the lack of space around the concatenation operator (left side, specifically)?

Comment (robbieaverill) [reply code/CwpInitialisationFilter.php]: Nope, it's not a rule in PSR-2

File: _config/documentconverter.yml (hunk 1 above)
Comment (robbieaverill) [reply _config/documentconverter.yml]: Updated in 29d8be2

File: code/extensions/RichLinksExtension.php [hunk 3]
```diff
@@ -1,50 +1,63 @@
...
+    public function RichLinks()
+    {
+        // Note:
+        // Assume we can use Regexes because the link will always be formatted
+        // in the same way coming from the CMS.
+
+        $content = $this->owner->value;
+
+        // Find all file links for processing.
+        preg_match_all('/<a.*href="\[file_link,id=([0-9]+)\].*".*>.*<\/a>/U', $content, $matches);
```
Comment (NightJar) [inline code/extensions/RichLinksExtension.php]: `@todo: don't use regex to parse html`?

Comment (robbieaverill) [reply code/extensions/RichLinksExtension.php]: Totally

**Pull request merged**